crypto/crypto.c

```diff
@@ -479,6 +722,19 @@ int crypto_pkcs7_signed_hash_verify(void *pkcs7, void *x509, unsigned char *hash
 {
     return mbedtls_pkcs7_signed_hash_verify((struct mbedtls_pkcs7 *)pkcs7, (mbedtls_x509_crt *)x509, hash, hash_len);
 }
+
+int crypto_pkcs7_generate_w_signature(unsigned char **pkcs7, size_t *pkcs7Size, const unsigned char *newData, size_t newDataSize, 
+    const char** crtFiles, const char** keyFiles,  int keyPairs, int hashFunct)
+{
+    return to_pkcs7_generate_signature(pkcs7, pkcs7Size, newData, newDataSize, crtFiles, keyFiles, keyPairs, hashFunct);
+}
+
+int crypto_pkcs7_generate_w_already_signed_data(unsigned char **pkcs7, size_t *pkcs7Size, const unsigned char *newData, size_t newDataSize, 
+    const char** crtFiles, const char** sigFiles,  int keyPairs, int hashFunct)
+{
+    return to_pkcs7_already_signed_data(pkcs7, pkcs7Size, newData, newDataSize, crtFiles, sigFiles, keyPairs, hashFunct);
+}
```

Comment:
is there any reason not to move the contents of those functions in to these new functions? then you can just get rid of the old ones.

Reply:
The code in `to_pkcs7_already_signed_data` uses lots of static structs and functions from `generate-pkcs7.c`. Also, I plan on adding these functions to my Mbedtls PR.